facet-reflect/src/peek/value.rs

```diff
@@ -61,7 +62,13 @@ impl<'mem> PeekValue<'mem> {
     /// `data` must be initialized and well-aligned, and point to a value
     /// of the type described by `shape`.
     pub(crate) unsafe fn unchecked_new(data: OpaqueConst<'mem>, shape: &'static Shape) -> Self {
-        Self { data, shape }
+        let scalar_type = ScalarType::try_from_shape(shape);
```

Comment:
Should I calculate the type here or in the getter itself?

Doing it here means it will be calculated at most once, but also when it's not used.
Doing it in `scalar_type` means it will be calculated every time it's called, but not when it's used.

Reply:
I vote "in the getter"